### DISCUSSION TITLE
`size` with `flip` for dynamic content: placement is sometimes incorrect

### DISCUSSION THREAD

**Discussion by eternalthinker** [floating-ui/floating-ui#2953] [Q&A]
I'm using the `size` middleware with `flip`, as described here: https://floating-ui.com/docs/size#using-with-flip

Depending on interactions within the content, my floating content can sometimes change in height. When height increases dynamically, sometimes `flip` doesn't seem to recognize that there is more space left on the other side (best fit), and sometimes it does.
Is there a pattern to this and a way to avoid it, or is this random due to a race condition?

Demo: https://codepen.io/eternalthinkerr/pen/XWwQqyb

https://github.com/floating-ui/floating-ui/assets/2195458/8ec804dc-e82e-4b79-8626-8f62989433d3

**Comment by atomiks** [comment]
I think I have noticed this sometimes, but since it seems like a valid bug, it would be better tracked as an issue with the repro provided!
